tool/lib/commands/release_helper.dart

```diff
@@ -0,0 +1,169 @@
+
+  @override
+  FutureOr? run() async {
+    final useCurrentBranch = argResults!['use-current-branch']!;
+    final currentBranchResult = await DevtoolsProcess.runOrThrow('git', [
+      'rev-parse',
+      '--abbrev-ref',
+      'HEAD',
+    ]);
+    final initialBranch = currentBranchResult.stdout.toString().trim();
+    String? releaseBranch;
+
+    try {
+      // Change the CWD to the repo root
+      Directory.current = pathFromRepoRoot("");
+      print("Finding a remote that points to flutter/devtools.git.");
+      final String devtoolsRemotes =
+          (await DevtoolsProcess.runOrThrow('git', ['remote', '-v'])).stdout;
+      final remoteRegexp = RegExp(
+        r'^(?<remote>\S+)\s+(?<path>\S+)\s+\((?<action>\S+)\)',
+        multiLine: true,
+      );
+      final remoteRegexpResults = remoteRegexp.allMatches(devtoolsRemotes);
+      final RegExpMatch devtoolsRemoteResult;
+
+      try {
+        devtoolsRemoteResult = remoteRegexpResults.firstWhere((element) =>
+            RegExp(r'flutter/devtools.git$')
+                .hasMatch(element.namedGroup('path')!));
+      } on StateError {
+        throw "ERROR: Couldn't find a remote that points to flutter/devtools.git. Instead got: \n$devtoolsRemotes";
+      }
+      final remoteOrigin = devtoolsRemoteResult.namedGroup('remote')!;
+
+      final gitStatusResult =
+          await DevtoolsProcess.runOrThrow('git', ['status', '-s']);
+      if (gitStatusResult.stdout.isNotEmpty) {
+        throw "Error: Make sure your working directory is clean before running the helper";
+      }
+
+      releaseBranch =
+          '_release_helper_release_${DateTime.now().millisecondsSinceEpoch}';
+
+      if (!useCurrentBranch) {
+        print("Preparing the release branch.");
+        await DevtoolsProcess.runOrThrow(
+            'git', ['fetch', remoteOrigin, 'master']);
+
+        await DevtoolsProcess.runOrThrow('git', [
+          'checkout',
+          '-b',
+          releaseBranch,
+          '$remoteOrigin/master',
+        ]);
+      }
+
+      print("Ensuring ./tool packages are ready.");
+      Directory.current = pathFromRepoRoot("tool");
+      await DevtoolsProcess.runOrThrow('dart', ['pub', 'get']);
+
+      Directory.current = pathFromRepoRoot("");
+
+      final currentVersionResult =
+          await DevtoolsProcess.runOrThrow('devtools_tool', [
+        'update-version',
+        'current-version',
+      ]);
+
+      final originalVersion = currentVersionResult.stdout;
+
+      print("Setting the release version.");
+      await DevtoolsProcess.runOrThrow('devtools_tool', [
+        'update-version',
+        'auto',
+        '--type',
+        'release',
+      ]);
+
+      final getNewVersionResult =
+          await DevtoolsProcess.runOrThrow('devtools_tool', [
+        'update-version',
+        'current-version',
+      ]);
+
+      final newVersion = getNewVersionResult.stdout;
+
+      final commitMessage = "Releasing from $originalVersion to $newVersion";
```

Comment:
nit: is the original version necessary? how about just "Prepare for release $newVersion"?